include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -220,7 +220,8 @@ class geometry_to_segment_or_box
         comparable_return_type cd_min1(0);
         point_iterator_type pit_min;
         seg_or_box_iterator_type it_min1 = seg_or_box_points.begin();
-        seg_or_box_iterator_type it_min2 = ++seg_or_box_points.begin();
+        seg_or_box_iterator_type it_min2 = it_min1;
+        ++it_min2;
```

Comment:
Following the explanation you gave to Barend for the issue in `buffered_piece_collection.hpp`, my suggestion would be to re-write lines 222 and 223 in the file as:
```
seg_or_box_iterator_type it_min1 = boost::const_begin(seg_or_box_points);
seg_or_box_iterator_type it_min2 = ++boost::const_begin(seg_or_box_points);
```
Please remember to write these in two lines, to be conformant with the 80-chars-per-line requirement.